### PR TITLE
Fix(refs:T31364): Edit fragment doesnt work

### PR DESCRIPTION
**Pull request by elmasdemos**
**Ticket:** https://yaits.demos-deutschland.de/T31364

Description: 

- save and finish a fragment text didnt work correctly

### PR Checklist

- [x] Link all relevant tickets
- [x] Move the tickets on the board accordingly

### CODE REVIEW

File: demosplan/DemosPlanCoreBundle/Controller/Statement/DemosPlanAssessmentStatementFragmentController.php [hunk 1]
```diff
@@ -391,7 +391,7 @@ public function editStatementFragmentAjaxAction(
             }
 
             // return current Version. Use ES to receive defined structure
-            $fragment = $this->statementHandler->getFragmentOfStatementES($updatedStatementFragment->getStatementId(), $fragmentId);
+            $fragment = $this->statementHandler->getFragmentOfStatement($updatedStatementFragment->getStatementId(), $fragmentId);
 
             return $this->renderJson($fragment, $returnCode, $success);
```
[the remark this answers is not in the document]
Comment (jalalifardemos) [reply demosplan/DemosPlanCoreBundle/Controller/Statement/DemosPlanAssessmentStatementFragmentController.php]: done in [dfcee31](https://github.com/demos-europe/demosplan-core/pull/2820/commits/dfcee311874b4497e830ff023a6d7b55fc8c54c4)

File: demosplan/DemosPlanCoreBundle/Logic/Statement/StatementHandler.php [hunk 2]
```diff
@@ -3825,6 +3825,21 @@ public function getFragmentOfStatementES($statementId, $fragmentId)
             ->filter(fn ($fragment) => $fragment['id'] === $fragmentId)->first(null, []);
     }
 
+    /**
+     * @param string $statementId
+     * @param string $fragmentId
+     *
+     * @return array|null;
+     */
+    public function getFragmentOfStatement($statementId, $fragmentId)
```
[the remark this answers is not in the document]
Comment (jalalifardemos) [reply demosplan/DemosPlanCoreBundle/Logic/Statement/StatementHandler.php]: done in [dfcee31](https://github.com/demos-europe/demosplan-core/pull/2820/commits/dfcee311874b4497e830ff023a6d7b55fc8c54c4)

File: demosplan/DemosPlanCoreBundle/Logic/Statement/StatementHandler.php [hunk 3]
```diff
@@ -3825,6 +3825,21 @@ public function getFragmentOfStatementES($statementId, $fragmentId)
             ->filter(fn ($fragment) => $fragment['id'] === $fragmentId)->first(null, []);
     }
 
+    /**
+     * @param string $statementId
+     * @param string $fragmentId
+     *
+     * @return array|null;
```
[the remark this answers is not in the document]
Comment (jalalifardemos) [reply demosplan/DemosPlanCoreBundle/Logic/Statement/StatementHandler.php]: done in [dfcee31](https://github.com/demos-europe/demosplan-core/pull/2820/commits/dfcee311874b4497e830ff023a6d7b55fc8c54c4)

File: demosplan/DemosPlanCoreBundle/Repository/StatementFragmentRepository.php [hunk 4]
```diff
@@ -60,6 +60,29 @@ public function get($entityId)
...
+     * @param string $statementId
+     */
+    public function findByStatement($statementId): ?array
+    {
+        try {
+            $query = $this->getEntityManager()->createQueryBuilder()
+                ->select('sf')
+                ->from(StatementFragment::class, 'sf')
+                ->where('sf.statement = :id')
+                ->setParameter('id', $statementId)
```
[the remark this answers is not in the document]
Comment (jalalifardemos) [reply demosplan/DemosPlanCoreBundle/Repository/StatementFragmentRepository.php]: done in [dfcee31](https://github.com/demos-europe/demosplan-core/pull/2820/commits/dfcee311874b4497e830ff023a6d7b55fc8c54c4)

File: demosplan/DemosPlanCoreBundle/Controller/Statement/DemosPlanAssessmentStatementFragmentController.php [hunk 5]
```diff
@@ -380,7 +380,7 @@ public function editStatementFragmentAjaxAction(
                 $updateData['mayChangeMetaData'] = false;
             }
             $updatedStatementFragment = $this->statementHandler->updateStatementFragment($fragmentId, $updateData, $isReviewer);
-            $returnCode = 200;
+            $returnCode = 400;
```
Comment (dresslerdemos) [inline demosplan/DemosPlanCoreBundle/Controller/Statement/DemosPlanAssessmentStatementFragmentController.php:383]: My intention was to return a 400 status code if `getFragmentOfStatement` returns `null`, not always. Also please use the constants provided by `Symfony\Component\HttpFoundation\Response`.

**Review by dresslerdemos — CHANGES_REQUESTED** [review]

File: demosplan/DemosPlanCoreBundle/Controller/Statement/DemosPlanAssessmentStatementFragmentController.php (hunk 5 above)
Comment (jalalifardemos) [reply demosplan/DemosPlanCoreBundle/Controller/Statement/DemosPlanAssessmentStatementFragmentController.php]: done in [74bd37a](https://github.com/demos-europe/demosplan-core/pull/2820/commits/74bd37a18126c403885926bcfd9aab97df4975a6)

File: demosplan/DemosPlanCoreBundle/Logic/Statement/StatementHandler.php [hunk 6]
```diff
@@ -3825,6 +3825,17 @@ public function getFragmentOfStatementES($statementId, $fragmentId)
...
 
+    /**
+     * @return array<string, mixed>|null;
+     */
+    public function getFragmentOfStatement(string $fragmentId): ?array
+    {
+        $fragment = $this->entityManager
+            ->getRepository(StatementFragment::class)->getAsArray($fragmentId);
+
+        return collect($fragment)->first(null, []);
```
[the remark this answers is not in the document]
Comment (jalalifardemos) [reply demosplan/DemosPlanCoreBundle/Logic/Statement/StatementHandler.php]: done in [74bd37a](https://github.com/demos-europe/demosplan-core/pull/2820/commits/74bd37a18126c403885926bcfd9aab97df4975a6)

File: demosplan/DemosPlanCoreBundle/Repository/StatementFragmentRepository.php [hunk 7]
```diff
@@ -60,6 +60,27 @@ public function get($entityId)
         }
     }
 
+    /**
+     * Get Entity by Id.
+     */
+    public function getAsArray(string $fragmentId): ?array
```
[the remark this answers is not in the document]
Comment (jalalifardemos) [reply demosplan/DemosPlanCoreBundle/Repository/StatementFragmentRepository.php]: done in [74bd37a](https://github.com/demos-europe/demosplan-core/pull/2820/commits/74bd37a18126c403885926bcfd9aab97df4975a6)

**Review by dresslerdemos — APPROVED** [review]
Backend looks ok now. If the frontend team approves that the new format is sufficient, then this PR can be merged.

File: client/js/components/statement/fragmentList/DpFragmentList.vue [hunk 8]
```diff
@@ -31,14 +31,14 @@
...
+import DpStatementFragment from './Fragment'
 import { mapMutations } from 'vuex'
-import StatementFragment from './Fragment'
 
 export default {
   name: 'DpFragmentList',
 
   components: {
-    'dp-statement-fragment': StatementFragment
+    'dp-statement-fragment': DpStatementFragment
```
Comment (spiess-demos) [inline client/js/components/statement/fragmentList/DpFragmentList.vue:41]: can now be simplified into:

```suggestion
    DpStatementFragment
```

File: client/js/components/statement/fragment/Version.vue [hunk 9]
```diff
@@ -86,13 +86,13 @@
...
 import { checkResponse, CleanHtml, dpApi, DpLoading, formatDate } from '@demos-europe/demosplan-ui'
-import FragmentStatus from './Status'
+import DpFragmentStatus from './Status'
 
 export default {
   name: 'DpFragmentVersion',
 
   components: {
-    'dp-fragment-status': FragmentStatus,
+    'dp-fragment-status': DpFragmentStatus,
```
Comment (spiess-demos) [inline client/js/components/statement/fragment/Version.vue:95]: can now be simplified to:

```suggestion
    DpFragmentStatus,
```

**Review by spiess-demos — DISMISSED** [review]
two nitpicky suggestions, otherwise lgtm!

File: client/js/components/statement/fragmentList/DpFragmentList.vue (hunk 8 above)
Comment (elmasdemos) [reply client/js/components/statement/fragmentList/DpFragmentList.vue]: done here https://github.com/demos-europe/demosplan-core/pull/2820/commits/3eb2e119e526abb83a5732cbd6339f8e55dd46d1

**Pull request merged**